infra/cdoc_web.Dockerfile

```diff
@@ -0,0 +1,21 @@
+# Base image
+FROM maven:3.8.6-openjdk-11 AS compiler
+# Copy encryption source
+WORKDIR /app/cdoc_jar
+COPY cdoc/cdoc_jar/pom.xml pom.xml
+COPY cdoc/cdoc_jar/src src
+COPY .git .git
+
+# Package project to jar
+RUN mvn package
+
+
+FROM openjdk:11.0.11-jre-slim
+
+EXPOSE 4444
+
+WORKDIR /app
+
+COPY --from=compiler /app .
```

Comment:
The point of 2 images is that the 2nd image has no trace of source code, you don't want other people getting access to your source through your image

Reply:
This source is open-source, and a jar package is easily decompilable.